### PR TITLE
Prac05: Add 01-03 | Presiyana Borisova

### PR DESCRIPTION
**Pull request by PresiyanaBB**
Presiyana Borisova
6 3MI0600240

### CODE REVIEW

File: SI/Sem.05/Pract.05/Task01.cpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#include <iostream>
+using namespace std;
+
+const int SIZE = 20;
+
+void findRepeatingNumber(int* arr, int size)
+{
+    int copyArr[SIZE];
+
+    for (int i = 0; i < size; i++)
```
Comment (Justsvetoslavov) [inline SI/Sem.05/Pract.05/Task01.cpp:10]: for (int i = 0; i < size - 1; i++)

File: SI/Sem.05/Pract.05/Task01.cpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    {
+        for (int j = 0; j < i; j++)
```
Comment (Justsvetoslavov) [inline SI/Sem.05/Pract.05/Task01.cpp:12]: for (int j = i + 1; j < size; j++)

File: SI/Sem.05/Pract.05/Task01.cpp [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            if (arr[i] == copyArr[j])
```
Comment (Justsvetoslavov) [inline SI/Sem.05/Pract.05/Task01.cpp:13]: arr[i] == arr[j]

File: SI/Sem.05/Pract.05/Task02.cpp [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+	else
+	{
+		for (int i = 0; i < size; i++)
+			for (int j = 0; j < size - i - 1; j++)
+				if (arr[j] < arr[j + 1])
+				{
+					int temp = arr[j];
+					arr[j] = arr[j + 1];
+					arr[j + 1] = temp;
```
Comment (GeorgiTerziev02) [inline SI/Sem.05/Pract.05/Task02.cpp:28]: Create a swap function to reuse some code

**Review by GeorgiTerziev02 — CHANGES_REQUESTED** [review]

File: SI/Sem.05/Pract.05/Task03.cpp [hunk 5]
```diff
@@ -0,0 +1,28 @@
+#include <iostream>
+using namespace std;
+
+void intToBinary(int num)
```
Comment (Justsvetoslavov) [inline SI/Sem.05/Pract.05/Task03.cpp:4]: We need to return the number.

We convert it to binary and return it.

Revise this task.

File: SI/Sem.05/Pract.05/Task01.cpp [hunk 6]
```diff
@@ -0,0 +1,45 @@
+#include <iostream>
+using namespace std;
+
+const int SIZE = 20;
+
+void findRepeatingNumber(int* arr, int size)
+{
+    int copyArr[SIZE];
```
[the remark this answers is not in the document]
Comment (Justsvetoslavov) [reply SI/Sem.05/Pract.05/Task01.cpp:8]: We do not need a copyArr.

We need to search in our given one without copies.

File: SI/Sem.05/Pract.05/Task02.cpp [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+{
+	for (int i = 0; i < size; i++)
+		cin >> arr[i];
+}
+
+int main()
+{
+	int arr[SIZE];
+	int evenArr[SIZE];
+	int oddArr[SIZE];
```
[the remark this answers is not in the document]
Comment (Justsvetoslavov) [reply SI/Sem.05/Pract.05/Task02.cpp:76]: That is correct.
However, as an initial solution it works (gotta point it out)

**Review by Justsvetoslavov — CHANGES_REQUESTED** [review]

**Comment by plamena-ilieva** [comment]
@Justsvetoslavov merging but please address the tasks.

**Pull request merged**